src/lando/headless_api/admin.py

```diff
@@ -24,18 +22,13 @@ class AutomationActionJobInline(ReadOnlyInline):
     _target_object = "actions"
     readonly_fields = ("action_type", "data", "order")
 
-    def formfield_for_dbfield(self, db_field: models.Field, **kwargs) -> forms.Field:
-        """
-        Forbid alteration of the JobAction list.
-        """
-        formfield = super(ReadOnlyInline, self).formfield_for_dbfield(
-            db_field, **kwargs
-        )
-        if db_field.name == "actions":
-            formfield.widget.can_add_related = False
-            formfield.widget.can_change_related = False
-            formfield.widget.can_delete_related = False
-        return formfield
+    def has_add_permission(self, request, obj=None) -> bool:  # noqa: ANN001
+        """Forbid addition of any action object from the inline interface."""
+        return False
+
+    def has_delete_permission(self, request, obj=None) -> bool:  # noqa: ANN001
+        """Forbid deletion of any action object from the inline interface."""
+        return False
```

Comment:
Nit: I think the types here are `request: HttpRequest, obj: AutomationAction | None = None`, we could add those and remove the `noqa`.

Reply:
The type-hinting of `request` is tricky (because Django doesn't use type-hints internally). We had a chat with @zzzeid the other day, and concluded that we should use `WSGIRequest` (which is a subclass of `HttpRequest` https://github.com/django/django/blob/0fe218842e0e396e3ab3982bd21227968a9e7fd8/django/core/handlers/wsgi.py#L56), as we use `uwsgi`.